test/verify/check-machines-dbus

```diff
@@ -575,6 +577,25 @@ class TestMachinesDBus(machineslib.TestMachines):
         b.wait_in_text("#vm-subVmTest1-network-1-select-type", "Bridge to LAN")
         b.select_from_dropdown("#vm-subVmTest1-network-1-select-type", "Virtual network")
         b.wait_present("#vm-subVmTest1-network-1-edit-dialog-save:disabled")
+        b.click(".modal-footer button:contains(Cancel)")
+
+        # Ensure that when the source of a NIC was removed we can still change it
+
+        # Redefine deleted networks and attach an interface with source a deleted network
+        m.execute("virsh net-define /tmp/net-default.xml && virsh net-start default && virsh attach-interface --persistent --type network --source default --domain subVmTest1")
+        m.execute("virsh net-destroy default && virsh net-undefine default")
+        m.execute("virsh net-define /tmp/net-test-network.xml && virsh net-start test_network")
+
+        # First shut of the VM otherwise the interface will not be the same in the live and config XML https://www.redhat.com/archives/libvir-list/2019-August/msg01034.html
+        b.click("#vm-subVmTest1-off-caret")
+        b.click("#vm-subVmTest1-forceOff")
+        b.wait_in_text("#vm-subVmTest1-state", "shut off")
+
+        # Try to edit the interface changing the source to a non deleted network
+        b.click("#vm-subVmTest1-network-1-edit-dialog")
+        b.select_from_dropdown("#vm-subVmTest1-network-1-select-source", "test_network")
+        b.click("#vm-subVmTest1-network-1-edit-dialog-save")
+        b.wait_in_text("#vm-subVmTest1-network-1-source", "test_network")
```

Comment:
If I understand it correctly, the bug which we are trying to fix here would happen only if we undefine a certain network and then create a new one with **different name**, right?
At the beginning of your addition, the subVmTest1 still had the vNIC with source "test_network". You deleted the "test_network", and then created a new one with the same name. So if a bugfix wouldn't work, this test would not catch it, because in either case (bugfix works or not), vNIC source would stay as "test_network" in both cases.

Reply:
The NIC source was default and not test network, see 'virsh attach-interface --persistent --type network --source default --domain subVmTest1' before the edit dialog gets open with the new test code addition.